test/jdk/sun/security/util/AlgorithmConstraints/DisabledAlgorithmPermits.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import sun.security.util.DisabledAlgorithmConstraints;
+
+import java.security.CryptoPrimitive;
+import java.security.KeyPairGenerator;
+import java.security.PrivateKey;
+import java.security.Security;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+/*
+ * @test
+ * @bug 8346129
+ * @modules java.base/sun.security.util
+ * @summary Check DisabledAlgorithmConstraints methods permit(Key) and
+ * permit(Set<CryptoPrimitive>, String, APS) with EdDSA & XDH keys
+ *
+ * @run main/othervm DisabledAlgorithmPermits Ed25519
+ * @run main/othervm DisabledAlgorithmPermits Ed448
+ * @run main/othervm DisabledAlgorithmPermits EdDSA
+ * @run main/othervm DisabledAlgorithmPermits X25519
+ * @run main/othervm DisabledAlgorithmPermits X448
+ * @run main/othervm DisabledAlgorithmPermits XDH
+ */
+
+public class DisabledAlgorithmPermits {
+    public static void main(String[] args) throws Exception {
+        String algorithm = args[0];
+        Security.setProperty("x", algorithm);
+
+        // Expected table are the expected results from the test
+        List<TestCase> expected = switch (algorithm) {
+            case "Ed25519" ->
+                Arrays.asList(
+                    new TestCase("EdDSA", false),
```

Comment:
As Sean mentioned in another comment, disabling "Ed25519" does not imply all EdDSA keys are not permitted. This means the result of `permits(primitives, algorithmName, parameters)` cannot be determined. That said, I noticed you've used `KeyUtil::getAlgorithm` in a lot of places. Can we guarantee that this `permits` method is never called on a family algorithm name? If so, we can get a definitive result.

Reply:
I believe my comment to Sean answers this question, but I'm not sure I understand the last question in your comment.  "never called on a family algorithm name".  The change is to make sure these two families return the curve name and not the family name (EdDSA & XDH).  But on the other side, someone using the family name of the disabled algorithm list would disable all curves.
The above test code is checking that this call ` permits(Set.of(CryptoPrimitive.SIGNATURE), "EdDSA", null)` will fail for a Ed25519 key because of the precedent set by KPG.

Reply:
We are talking about the same in multiple comments now.

In this case, if both `permits(SIGNATURE, "EdDSA", null)` and `permits(SIGNATURE, key)` are called, it's safe to bypass the 1st check as long as the 2nd one blocks the key. So it's not necessary to cover "EdDSA" when only "Ed25519" is disabled.